torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -508,8 +508,17 @@ def type_check_node(self, n: Node):
         if n.op == 'placeholder':
             return n.type
 
+        elif n.op == 'get_attr':
+            t = self.traced.get_parameter(n.target)
+            if isinstance(t.data, torch.Tensor):
+                n.type = TensorType(t.data.shape)
+            return n.type
+
         elif n.op == 'call_function':
-            if n.target in _INFERENCE_RULES:
+            if n.target in _INFERENCE_RULES and n.target == getattr:
```

Comment:
Is `getattr` always going to be in `_INFERENCE_RULES`? If so, could just make this condition `if n.target == getattr` and then `assert getattr in _INFERENCE RULES` the branch. That would simplify reading the logic here.